File: my-folder/1802-number-of-students-unable-to-eat-lunch/solution.java
class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
        int ptr = 0;
        Queue<Integer> queue = new LinkedList<>();
        
        for(int i : students){
            queue.offer(i);    
        }
        
        int count = 0;
        int len = students.length;
        while(!queue.isEmpty()){
            int s = queue.poll();
            if(sandwiches[ptr] == s){
                len--;
                ptr++;
                count = 0;
            } else {
                queue.offer(s);
                count++;
                if(count == len) return len;
            }
            
        }
                
    
        
        return 0;
    }
}

File: my-folder/1802-number-of-students-unable-to-eat-lunch/solution.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        yum = [
        sum(sand == 0 for sand in students),
        sum(sand == 1 for sand in students) # meh but ok
        ]

        ans = len(students)
        
        for sandwich in sandwiches:
            if yum[sandwich] == 0: return ans
            yum[sandwich] -= 1
            ans -= 1
        
        return 0
                
